docs/_docs/monitoring-metrics/performance-statistics.adoc

```diff
@@ -0,0 +1,85 @@
+
+== Collecting statistics
+
+To start and stop collecting statistics use the link:#jmx[JMX interface].
+
+Each node collects performance statistics to the file in binary format. This file is placed under `Ignite_work_directory/perf_stat/` directory.
+The name mask is `node-{nodeId}-{index}.prf`.
+
+Performance statistics files are used to build the report offline. Stop collecting statistics before creating the report.
+
+Nodes use off-heap buffer to temporarily store serialized statistics. The writer thread flushes buffer to the file when the flush size is reached. See the link:#system-properties[properties] section for details.
+
+Each start of collecting statistics will create a new file on nodes. The index in the file name will be increased if such file exists:
+
+    node-faedc6c9-3542-4610-ae10-4ff7e0600000.prf
+    node-faedc6c9-3542-4610-ae10-4ff7e0600000-1.prf
+    node-faedc6c9-3542-4610-ae10-4ff7e0600000-2.prf
+
+== Building the report
+
+Ignite provides a tool to generate the report from performance statistics files. The tool is published in the ignite-extensions repository: https://github.com/apache/ignite-extensions/tree/master/modules/performance-statistics-ext/[performance-statistics-ext, windows="_blank"].
+
+To build the performance report follow:
+
+1. Collect performance statistics files from all nodes under an empty directory. For example:
+
+    path_to_files
+        ├── node-162c7147-fef8-4ea2-bd25-8653c41fc7fa.prf
+        ├── node-7b8a7c5c-f3b7-46c3-90da-e66103c00001.prf
+        └── node-faedc6c9-3542-4610-ae10-4ff7e0600000.prf
+
+2.  Run the script from the release package of the tool:
+
+        performance-statistics-tool/build-report.sh path_to_files
+
+The performance report will be created in the new directory under the performance statistics files path:
+`path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see the report.
+
+== Management
+
+=== JMX
+
+You can manage the performance statistics collection via the `PerformanceStatisticsMBean` interface:
+
+[cols="1,2",opts="header"]
+|===
+|Method | Description
+|start() | Start collecting performance statistics in the cluster.
+|stop() | Stop collecting performance statistics in the cluster.
+|started() | True if collecting performance statistics is started.
+|===
+
+=== System properties
+
+[cols="2,1,1,3",opts="header"]
+|===
+|Property | Type | Default Value | Description
+|IGNITE_PERF_STAT_FILE_MAX_SIZE | Long | 32 Gb | Performance statistics maximum file size in bytes. Performance statistics will be stopped when the size exceeded.
+|IGNITE_PERF_STAT_BUFFER_SIZE | Integer | 32 Mb | Performance statistics off heap buffer size in bytes.
+|IGNITE_PERF_STAT_FLUSH_SIZE | Integer | 8 Mb | Performance statistics minimal batch size to flush in bytes.
+|IGNITE_PERF_STAT_CACHED_STRINGS_THRESHOLD | Integer | 1024 | Performance statistics maximum cached strings threshold. String caching will stop on threshold excess.
```

Comment:
'Maximum performance statistics cached strings threshold. String caching will be stopped, when the threshold is exceeded.

Reply:
Fixed.